packages/components/pie-card/src/index.ts

```diff
@@ -117,6 +117,62 @@ export class PieCard extends LitElement implements CardProps {
         return `padding: ${paddingCSS}`;
     }
 
+    /**
+     * Handles the slot change event and applies/removes opacity to images based on the `disabled` state.
+     *
+     * @private
+     */
+    private handleSlotChange () {
+        this.updateImagesOpacity();
+    }
+
+    /**
+     * Updates opacity of all images (slotted and non-slotted) based on the `disabled` property.
+     */
+    private updateImagesOpacity () {
+        // Handle slotted images
+        const slot = this.shadowRoot?.querySelector('slot');
+
+        if (slot) {
+            const slottedElements = slot.assignedElements({ flatten: true });
```

Comment:
Can we use one of the built-in decorators to handle this, like [queryAssignedElements](https://lit.dev/docs/api/decorators/#queryAssignedElements)?

Reply:
Fixed.